analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+- Any clusters with high endothelial marker gene expression also correspond to clusters that are predominantly "endothelial cell" in the heatmaps, so they will be labeled as "macrophage". 
+- There is one cluster in `SCPCL000822` that has high mesenchymal gene expression and is predominantly chondrocytes so we will label this cell as "chondrocytes". 
+- All other clusters with high mesenchymal gene expression appear to have a mix of chondrocytes, fibroblasts, and smooth muscle cells so we will just label these as "mesenchymal-like cells". 
+
+```{r}
+# get umap embeddings for plotting later
+umap_df <- umap_df |>
+  dplyr::select(library_id, barcodes, UMAP1, UMAP2) |>
+  dplyr::distinct()
+
+# assign clusters based on marker gene expression and cell types that are prominent in that cluster
+# use the jaccard matrices to help guide assignments
+cluster_umap_df <- cluster_classification_df |>
+  dplyr::left_join(umap_df, by = c("library_id", "barcodes")) |>
+  dplyr::filter(nn_param == "k_25") |>
+  dplyr::mutate(
+    cluster_celltype = dplyr::case_when(
+      library_id == "SCPCL000822" & cluster %in% c(1, 2, 4, 5, 9) ~ "tumor",
+      library_id == "SCPCL000822" & cluster == 7 ~ "endothelial cell",
+      # cluster 3 has predominantly chondrocytes and has MSC marker gene expression
+      library_id == "SCPCL000822" & cluster == 3 ~ "chondrocyte",
+      # clusters with high "immune" marker gene expression are macrophage
+      library_id == "SCPCL000822" & cluster == 6 ~ "macrophage",
+      # cluster 8 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000822" & cluster == 8 ~ "mesenchymal-like cell",
+      library_id == "SCPCL000824" & cluster %in% c(1, 2, 3, 4, 5, 6, 11) ~ "tumor",
+      library_id == "SCPCL000824" & cluster == 8 ~ "endothelial cell",
+      # cluster 10 has high immune markers and is mostly macrohpages
+      library_id == "SCPCL000824" & cluster == 10 ~ "macrophage",
+      # cluster 7 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000824" & cluster == 7 ~ "mesenchymal-like cell",
+      # if they didn't have high expression of any markers, we label them as unknown
+      .default = "unknown"
+    ),
+    # make tumor first
+    cluster_celltype = forcats::fct_relevel(cluster_celltype, "tumor", after = 0)
+  )
+```
+
+```{r}
+# look at cell type assignments on the UMAP
+ggplot(cluster_umap_df, aes(x = UMAP1, y = UMAP2, color = cluster_celltype)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  facet_grid(cols = vars(library_id)) +
+  theme(aspect.ratio = 1) +
+  guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5))) +
+  scale_color_brewer(palette = "Dark2")
+```
+
+We can also validate that these cells do in fact have higher expression of the marker genes as compared to other cells by plotting the marker gene expression on UMAPs. 
```

Comment:
It's hard to interpret these plots since they paired by cell type, not by sample. Is it possible to instead have two plots - one for each sample and faceted by cell type? (right now it's 3 plots, one for each cell type but faceted by sample)